### PR TITLE
Feature/check auth in header

### PR DESCRIPTION
**Pull request by edbrett**
Adds a service for checking if the user is logged in:
- gets token from local storage or url (if coming from a social login)
- checks auth
- cleans URL of token
- manages state of My GFW link

### CODE REVIEW

File: src/components/header/index.js [hunk 1]
```diff
@@ -50,10 +53,53 @@ class Header extends PureComponent {
...
+      delete query.token;
+      const cleanQuery = query && qs.stringify(query);
+      window.history.pushState(
+        {},
+        '',
+        `${window.location.pathname}${cleanQuery ? `?${cleanQuery}` : ''}`
+      );
     }
 
+    checkLoggedIn(token)
```
Comment (aabdaab) [inline src/components/header/index.js:89]: now, even if there's no token we're sending the request so we could add an `if (token)` here

Comment (aabdaab) [reply src/components/header/index.js:89]: ah, `checkLoggedIn` has this condition already

**Comment by aabdaab** [comment]
I think it's a correct approach; we're doing a fetch when the component mounts and let the associated components re-render once the browser is available. `componentDidMount` and `componentDidUpdate` will be triggered only on the client to it seems like the right place to do these things 🤔

File: src/components/header/index.js (hunk 1 above)
Comment (edbrett) [reply src/components/header/index.js:89]: Good spot. Indeed it should only be done if token exists.

Comment (edbrett) [reply src/components/header/index.js]: Updated

**Comment by edbrett** [comment]
Updated! I think its the simplest approach 🌴

**Pull request merged**